iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/WeakHashAlgorithmsCheck.java

```diff
@@ -0,0 +1,157 @@
+    private final List<String> parameters;
+    private final TextRange textRange;
+
+    public static List<Command> parse(List<Argument> arguments) {
+      return splitArgumentsPerGroup(arguments).stream()
+        .map(Command::new)
+        .collect(Collectors.toList());
+    }
+
+    /**
+     * Divide the provided list of string per separator into sublist.
+     * The sublist are wrapped up in GroupStringWithTextRange to keep the original text range with them.
+     * <pre>
+     *   ["exe1", "param1", "&&", "exe2", "param2"]
+     *   -> [["exe1", "param1"], ["exe2", "param2"]]
+     * </pre>
+     */
+    private static List<GroupStringWithTextRange> splitArgumentsPerGroup(List<Argument> arguments) {
+      List<String> resolvedArguments = resolveArguments(arguments);
+      int[] indexesSeparator = Stream.of(IntStream.of(-1),
+          IntStream.range(0, resolvedArguments.size())
+            .filter(i -> SEPARATOR.contains(resolvedArguments.get(i)))
+        ).flatMapToInt(s-> s).toArray();
+      return IntStream.range(0, indexesSeparator.length)
+        .mapToObj(i -> {
+          int indexFrom = indexesSeparator[i]+1;
+          int indexTo = i + 1 < indexesSeparator.length ? indexesSeparator[i + 1] : arguments.size();
+          return new GroupStringWithTextRange(arguments.subList(indexFrom, indexTo), resolvedArguments.subList(indexFrom, indexTo));
+        })
+        .collect(Collectors.toList());
+    }
+
+    private static List<String> resolveArguments(List<Argument> arguments) {
+      return arguments.stream()
+        .map(arg -> ArgumentResolution.of(arg).value())
+        .collect(Collectors.toList());
+    }
+
+    public Command(GroupStringWithTextRange elements) {
+      this.executable = elements.strings.isEmpty() ? null : elements.strings.get(0);
+      this.options = new HashMap<>();
+      this.parameters = new ArrayList<>();
+      this.textRange = elements.textRange;
+
+      elements.strings.stream()
+        .skip(1)
+        .forEach(this::processElement);
+    }
+
+    private void processElement(@Nullable String element) {
+      if (element != null && element.startsWith("-")) {
+        addOption(element);
+      } else {
+        this.parameters.add(element);
+      }
+    }
+
+    private void addOption(String option) {
+      option = option.replaceAll("^-++", "");
+      String[] split = option.split("=", 2);
+      if (split.length == 2) {
+        options.put(split[0], split[1]);
+      } else {
+        options.put(split[0], null);
+      }
+    }
+
+    private static class GroupStringWithTextRange {
+      private final List<String> strings;
+      private final TextRange textRange;
+
+      public GroupStringWithTextRange(List<Argument> arguments, List<String> resolvedArguments) {
+        this.textRange = arguments.isEmpty() ? null : TextRanges.merge(arguments.stream().map(HasTextRange::textRange).collect(Collectors.toList()));
+        this.strings = resolvedArguments;
+      }
+    }
```

Comment:
Baby step to retain the `textRange` of resolved argument + report issue on the whole `Command` block.
Next baby step is to only report the issue on related element : `openssl` keyword + `-md5` option for example.
A solution could be to keep the Argument object reference in ArgumentResolution, so that we can resolve argument, get/check the value and later access the original Argument object to report the issue on it.
WDYT ?

Reply:
Do you think [SONARIAC-647](https://sonarsource.atlassian.net/browse/SONARIAC-647) is the solution for this? PR is ready for this.

Reply:
Indeed it would totally solve that ! Let me know if/when you want me to review this PR, I don't see it in the list currently.

Reply:
https://github.com/SonarSource/sonar-iac/pull/613